src/currency-field/CurrencyField.ts

```diff
@@ -0,0 +1,377 @@
+         * Results can be the following based on the locale provided.
+         * ["$", "1", ",", "000", ".", "00"]
+         * ["1", ".", "000", ",", "00", " ", "$"]
+         */
+        let currencyPartsMap = [];
+        try {
+            currencyPartsMap = new Intl.NumberFormat(this.locale, {
+                style: 'currency',
+                currency: this.currency,
+                currencyDisplay: 'narrowSymbol'
+            })
+                .formatToParts(1000.0)
+                .map((c) => c.value);
+        } catch (e) {
+            console.error('invalid locale or currency provided, fallback will be used', e);
+            currencyPartsMap = new Intl.NumberFormat('en-US', { style: 'currency', currency: 'USD', currencyDisplay: 'narrowSymbol' })
+                .formatToParts(1000.0)
+                .map((c) => c.value);
+        }
+        // Get the currency symbol assuming it is the value at the first index of the array.
+        const currencySymbol = currencyPartsMap[0];
+
+        // Check to see if value at first index of the array is a number Some locales will have the currency symbol at the last index.
+        if (parseInt(currencySymbol)) {
+            this._currencyFormatSeparator = currencyPartsMap[CURRENCY_SEPARATOR.AMOUNT];
+            this._currencyCentsSeparator = currencyPartsMap[CURRENCY_SEPARATOR.CENTS];
+            this._currencySymbol = currencyPartsMap[currencyPartsMap.length - 1];
+        } else {
+            this._currencyFormatSeparator = currencyPartsMap[CURRENCY_SEPARATOR.INVERSE_AMOUNT];
+            this._currencyCentsSeparator = currencyPartsMap[CURRENCY_SEPARATOR.INVERSE_CENTS];
+            this._currencySymbol = currencySymbol;
+        }
+    }
+
+    _parseAmount(value: string): number {
+        let cleanValue = '';
+
+        for (let i = 0; i < value.length; i++) {
+            const character = value.charAt(i);
+            if (/\d/.test(character)) {
+                cleanValue += character;
+            }
+        }
+
+        if (cleanValue) {
+            return parseInt(cleanValue);
+        } else {
+            return null;
+        }
+    }
+
+    _parseCents(value: string): string {
+        let cleanValue = '';
+        for (let i = 0; i < value.length; i++) {
+            const character = value.charAt(i);
+            if (/\d/.test(character)) {
+                cleanValue += character;
+            }
+        }
+        return cleanValue;
+    }
+
+    _formatToCurrency(preFormattedValue: number): string {
+        if (!preFormattedValue) {
+            return '';
+        }
+        return this._currencyFormat.format(preFormattedValue);
+    }
+
+    // Format the internal value to a float.
+    _formatToFloat(formattedValue: string): string | number {
+        if (formattedValue.length > 0) {
+            const preFloatAllReplace = formattedValue.replaceAll(this._currencyFormatSeparator, '');
+            return parseFloat(preFloatAllReplace);
```

Comment:
What happens if the cent separator is not a decimal point? Does this take that into consideration?